model/model.js

```diff
@@ -1,8 +1,202 @@
 const db = require("../db/connection.js");
 
-exports.selectTopics = () => {
+const selectTopics = () => {
     return db.query("SELECT * FROM topics;").then((selectTopicsQueryResult) => {
         const selectTopicsResponseObject = {topics: selectTopicsQueryResult.rows};
         return selectTopicsResponseObject;
     });
-};
\ No newline at end of file
+};
+
+const selectArticles = (urlParams) => {
+    const lowercaseParams = {};
+    Object.keys(urlParams).forEach((element, index) => {
+        // default to lowercase
+        lowercaseParams[element.toLowerCase()] = urlParams[element].toLowerCase();
+    });
+
+
+    let isError = false;
+    let isErrorString = "";
+    let sortBy = "created_at";
+    let topicBool = false;
+    let topic = "";
+    let order = "desc"
+
+
+    if (urlParams) {
+                
+        if (lowercaseParams["sort_by"]) {
+            // check if a valid column was specified
+            const columnsList = ["title","topic","author","body","created_at","votes"];
+            if (columnsList.includes(lowercaseParams["sort_by"])) {
+                sortBy = lowercaseParams["sort_by"];
+            } else {
+                isError = true;
+                isErrorString = "invalid parameter was specified in sort_by";
+            };
+        };
+
+        if (lowercaseParams["order"]) {
+
+
+            const orderListAsc = ["asc",  "ascending",  "a"];
```

Comment:
why have you allowed people to submit 3 options for ASC and DESC, but this behaviour hasn't been tested